From 5bcc9c806ac0a8cb61030774b13944f4adc0859c Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Thu, 8 Jun 2023 20:43:29 +0200
Subject: [PATCH] Fix building by lisp without endian features

Some lisp may not have little-endian or big-endian in features which
leads to trigger that asset that shouldn't happened because target
features contains it and it should downgrade to bvref-64/bvref-32
instead of native-bvref-word.

This issue was introduced at 1ec9a7270aeb2f2b4d7d14919e323b44ee117be8
---
 src/cold/shebang.lisp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git src/cold/shebang.lisp src/cold/shebang.lisp
index 4edafcbf8..313036fc5 100644
--- src/cold/shebang.lisp
+++ src/cold/shebang.lisp
@@ -34,7 +34,9 @@
 (defun compatible-vector-raw-bits () ; T if the host and target match on word size and endianness
   (flet ((endianness (features)
            (let ((result (intersection '(:little-endian :big-endian) features)))
-             (assert (and result (not (cdr result))))
+             ;; some lisp implementation may not have little-endian / big-endian
+             ;; features which shouldn't trigger that assert
+             (assert (or (not result) (and result (not (cdr result)))))
              (car result)))
          (wordsize (features)
            (if (member :64-bit features) 64 32)))
-- 
2.41.0

